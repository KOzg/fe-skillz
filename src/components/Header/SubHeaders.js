import { React, Fragment, useState, useEffect, useRef, memo } from 'react';
import { useSelector } from 'react-redux';
import { Col } from 'react-bootstrap';
import {isMobile} from 'react-device-detect';
import classes from './../../styles/SubHeaders.module.scss';
import classnames from 'classnames';

function SubHeaders ({headerId}) {
    const [isHovered, setIsHovered] = useState(null);
    const [subs, setSubs] = useState([]);
    const {subHeaders} = useSelector((state) => state.headers);

    useEffect(() => {
        setSubs(subHeaders && subHeaders.filter(sub => {return sub.headerId === headerId}));
    }, [])

    const navClassNames = isMobile ? 
                            classes.sub_header__mobile : 
                            classes.sub_header;
    const getColClasses = (index) => {
         return {
            [classes.sub_header__mobile__subName]: isMobile,
            [classes.sub_header__subName]: !isMobile,
            [classes.sub_header__subName__hovered]: !isMobile && isHovered === index
        }
     }

    const GetSubHeaders = memo(() => {
        return (
        <nav className={navClassNames}>
            {subs && subs.length > 0 && subs.map((sub, index) => {
                return (
                    <Fragment key={sub.id}>
                        <Col 
                            onMouseEnter={() => {setIsHovered(index)}}
                            onMouseLeave={() => {setIsHovered(null)}}
                            className={classnames(getColClasses(index))}>
                                <span>{sub.name}</span>
                        </Col>
                    </Fragment>
                )})}
        </nav>
        )
    });

    return ( 
        <GetSubHeaders />
    );
}

export default SubHeaders;