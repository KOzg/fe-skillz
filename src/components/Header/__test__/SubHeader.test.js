import React from 'react';
import SubHeaders from './../SubHeaders';
import * as reactRedux from 'react-redux';
import { create } from 'react-test-renderer';

const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');

beforeEach(() => {
    useSelectorMock.mockClear();
  })

  let component;

describe('SubHeader', () => {

    it('matches the snapshot', () => {
        useSelectorMock.mockReturnValue({});
        component = create(<SubHeaders headerId={0} />);
        expect(component.toJSON).toMatchSnapshot();
    })

    it('component shouldnt crash even if no sub headers', () => {
        useSelectorMock.mockReturnValue({});
        component = create(<SubHeaders headerId={0}/>).toJSON();
        expect(component.type).toBe('nav');
        expect(component.props.className.includes('__subNav')).toBe(true);
        expect(component.children).toBe(null);
    });
  })