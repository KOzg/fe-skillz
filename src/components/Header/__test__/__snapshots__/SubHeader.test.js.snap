// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader matches the snapshot 1`] = `[Function]`;
